containers/Dockerfile.buildenv

```diff
@@ -163,13 +302,28 @@ RUN wget https://github.com/doxygen/doxygen/archive/Release_1_9_3.tar.gz -O doxy
         -D CMAKE_BUILD_TYPE=Release .. \
     && make $PARALLEL_MAKE_ARG
 
+# WARNING: We now copy the things from the "xbuild" container/image into the
+#          xbuild-test container to make sure the copying works. We do the
+#          actual copy into the spectre container below.
+#
 FROM base AS xbuild-test
+ARG UBUNTU_VERSION=22.04
 COPY --from=xbuild /root/ccache/build/ccache /usr/local/bin
 RUN ccache --version
 COPY --from=xbuild /root/doxygen/build/bin/doxygen /usr/local/bin
 RUN doxygen --version
+COPY --from=xbuild /usr/local/bin/cmake /work/
+RUN /work/cmake --version \
+    && if [ ${UBUNTU_VERSION} = 18.04 ]; then \
+    rm /usr/local/bin/ccache /usr/local/bin/doxygen; \
+    else \
+    mv /work/cmake /usr/local/bin; \
+    fi
 
 
+# WARNING: We are now back in the spectre container but we have 2, one for each
```

Comment:
The docker build doesn't know anything about the other architecture it's building for. I'd delete the "we have 2" part of this comment.

Reply:
Sorry, how do we not have two? Below it says `base-amd64` and `base-arm64`

Reply:
From the two stages `base-amd64` and `base-arm64` only the arch we're actually building is used. The line `FROM base-${TARGETARCH} AS dev` selects it. This is effectively an `if` in the Dockerfile (it's annoying that the Dockerfile syntax doesn't make that easier).

Reply:
Ah okay. What do you think of the wording:

We are now back in the spectre container. We create two base images for the 2 possible target architectures and then select the one we need as the `dev` container below. This is essentially an `if` in the Dockerfile.

Reply:
Sounds good 👍